framework/src/utils/ExecFlagEnum.C

```diff
@@ -0,0 +1,96 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ExecFlagEnum.h"
+#include "MooseError.h"
+
+ExecFlagEnum::ExecFlagEnum() : MultiMooseEnum() {}
+ExecFlagEnum::ExecFlagEnum(const MultiMooseEnum & other) : MultiMooseEnum(other) {}
+ExecFlagEnum::ExecFlagEnum(const ExecFlagEnum & other) : MultiMooseEnum(other) {}
+
+void
+ExecFlagEnum::addAvailableFlags(const ExecFlagType & flag)
+{
+  addEnumerationItem(flag);
+}
+
+void
+ExecFlagEnum::removeAvailableFlags(const ExecFlagType & flag)
+{
+  if (find(flag) == _items.end())
+    mooseError("The supplied item '",
+               flag,
+               "' is not an available enum item for the "
+               "MultiMooseEnum object, thus it cannot be removed.");
+  else if (contains(flag))
+    mooseError("The supplied item '", flag, "' is a selected item, thus it can not be removed.");
+
+  _items.erase(flag);
+}
+
+std::string
+ExecFlagEnum::getDocString() const
+{
+  std::string doc("The list of flag(s) indicating when this object should be executed, the "
+                  "available options include \'");
+  for (const std::string & name : getNames())
+    doc += name + "', '";
+  doc.erase(doc.end() - 4, doc.end());
```

Comment:
Umm....   Try using the new container stringify method. It should clean this whole mess up.